### ISSUE TITLE
Is this one-way only (iTunes playlist --> folder)?

### ISSUE THREAD

**Issue by brianroach** [mofirouz/iTunesPlaylistTrackSyncer#1]
Would it be possible to do the opposite? 

Sync folder (or, better yet, a folder hierarchy) to playlist folders & playlists? Perhaps implementing a naming structure like Vlad has here with this applescript: http://vladalexa.com/scripts/applescript/itunes/

Scenario:

	• Samples <--------------------------- (index 0)
      • House <--------------------------- (index 1)  
              • Sample Pack A <----------- (index 2)
              • Sample Pack B
              • Sample Pack C
      • Deep House
              • Sample Pack A
              • Sample Pack B
              • Sample Pack C
      • Techno
              • Sample Pack A
              • Sample Pack B
              • Sample Pack C
              • Sample Pack D
              • Sample Pack E
              • Sample Pack F
      • Trance
              • Sample Pack A
              • Sample Pack B
              • Sample Pack C

Would like it to do this:  
- For each Playlist Folder that is at 'index 0' that doesn't exist in iTunes, create Playlist Folder with matching name.  
- For each Playlist Folder that is at 'index 1' that doesn't exist in iTunes, create Playlist Folder with matching.  
- For each Playlist that is at 'index 2' that doesn't exist in iTunes, create Playlist with matching name.

Thoughts?

**Comment by mofirouz** [comment]
Sadly this won't be possible.

As far as I'm aware, `iTunes Music Library.xml` is generated by iTunes but not consumed by it, so changes are strictly one way from iTunes. iTunes uses the `.itl` binary file to store its playlist information. This isn't even possible via the Obj-C iTunes Framework as it will give read-only access.

You could (potentially) try an AppleScript way where it interacts with the visual menus of iTunes to import a M3U file playlist.

**Comment by mofirouz** [comment]
Also, FYI there is a bug in my go-code currently that I need to fix. URL decoding is non-existence in Go :disappointed:

**Comment by mofirouz** [comment]
I've just updated the code and released 1.0 of the binary. Fixed the issues I mentioned earlier

**Comment by brianroach** [comment]
@mofirouz I just tested this out with my iTunes library that has over 500 GB of music and found that it IS, in fact, possible to have iTunes consume the `iTunes Music Library.xml` files to completely restore the library.

With my iTunes settings set to NOT copy media to library and to NOT keep media folder organized, what I did was:
- File > Library > Export Library > then saved it as `Library.xml`to desktop
- Quit iTunes
- Delete `iTunes Library.xml` & `iTunes Library.it` from ~/Music/iTunes/
- Relaunch iTunes to find an empty library
- Go back into Preferences > Advanced & un-check `Keep iTunes folder organized` as well as `Copy media to iTunes folder...` (THIS PART IS IMPORTANT TO MAKE SURE YOU DON'T DUPLICATE YOUR ENTIRE MUSIC COLLECTION)
- File > Library > Import Library > select the `Library.xml` you created on your desktop & wait for the import process to finish. (Note: I noticed that when `Keep iTunes folder organized` as well as `Copy media to iTunes folder...` are un-checked iTunes doesn't display a progress bar during the import, whereas it does when these are checked & it is copying tracks to your media folder one by one)
- Whole library is restored - playlists and all! One thing tho is that the playlist folders to not restore correctly. Rather, they are re-created, but are empty. All the playlists they contained tho restore tho.

Game changing!
